packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -899,17 +960,62 @@ export class ProjectService {
             warehouseClient,
             granularity,
         );
+
         const compiledMetricQuery = compileMetricQuery({
             explore: exploreWithOverride,
             metricQuery,
             warehouseClient,
         });
-        return buildQuery({
+
+        const buildQueryResult = buildQuery({
             explore: exploreWithOverride,
             compiledMetricQuery,
             warehouseClient,
             userAttributes,
         });
+
+        return buildQueryResult;
+    }
+
+    /**
+     * Modifies a compiled metric query to remove any references to table calculations, and
+     * returns the isolated portions separately.
+     *
+     * This is a temporary approach to avoid poluting the query compiler while rolling-out
+     * improvements to table calculations handling.
+     */
+    private static isolateTableCalculationsFromCompiledMetricsQuery(
+        compiledMetricQuery: CompiledMetricQuery,
+    ): {
+        originalMetricQuery: CompiledMetricQuery;
+        compiledMetricQuery: CompiledMetricQuery;
+        tableCalculationFilters?: FilterGroupItem;
+        tableCalculations: TableCalculation[];
+        compiledTableCalculations: CompiledTableCalculation[];
+    } {
+        const {
+            filters,
+            tableCalculations,
+            compiledTableCalculations,
+            ...otherProps
+        } = compiledMetricQuery;
+
+        return {
+            originalMetricQuery: compiledMetricQuery,
+            compiledMetricQuery: {
```

Comment:
what about `compiledMetricQueryWithoutTableCalculations` ? That would make the code in the calling function clearer to me